src/components/FullscreenModal/hooks/useModal.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import { isNull } from 'ramda-adjunct';
+
+import { DSContext } from '../../../theme/DSProvider/DSProvider';
+
+const createEl = (id: string): HTMLElement => {
+  const el = document.createElement('div');
+  el.setAttribute('id', id);
+  document.body.appendChild(el);
+
+  return el;
+};
+
+const getContainer = (id: string): HTMLElement =>
+  document.getElementById(id) || createEl(id);
+
+export const useModal = (): React.FC | null => {
+  const { portalsContainerId } = useContext(DSContext);
+  const containerRef = useRef(getContainer(portalsContainerId));
+
+  useEffect(() => {
+    if (isNull(containerRef.current)) return null;
+    const { current } = containerRef;
+
+    current.style.position = 'absolute';
+    current.style.zIndex = '1250';
+    current.style.top = '0';
+    current.style.left = '0';
+
+    return () => {
+      current.style.position = 'static';
+      current.style.zIndex = 'auto';
+      current.style.top = 'auto';
+      current.style.left = 'auto';
+    };
+  }, [portalsContainerId]);
+
+  return ({ children }) => {
+    if (isNull(containerRef.current)) return null;
+    return createPortal(children, containerRef.current);
```

Comment:
since we're only passing it to the portal I think the terminology could be changed to singular ```child``` as it's in the documentation.

Reply:
Well, it actually returns functional component and `children` is `children` property not sure if it make sense alias it just for fun :D

Reply:
yep, that was exactly the argument I was having with myself as well 😂